includes/sanitizers/trait-amp-noscript-fallback.php

```diff
@@ -52,6 +53,13 @@ protected function initialize_noscript_allowed_attributes( $tag ) {
 				}
 			}
 		}
+
+		// Remove attributes which are likely to cause styling conflicts, as the noscript fallback should get treated like it has fill layout.
+		unset(
+			$this->noscript_fallback_allowed_attributes[ Attribute::ID ],
+			$this->noscript_fallback_allowed_attributes[ Attribute::CLASS_ ],
```

Comment:
Removing the class attribute has inadvertently caused `noscript>img` elements of custom logos to be detected as hero images, and because of that the AMP page fails to render due to a PHP fatal error:

```
[06-Apr-2021 08:46:38 UTC] PHP Warning:  Couldn't fetch AmpProject\Dom\Element. Node no longer exists in /app/public/content/plugins/amp/vendor/ampproject/amp-toolbox/src/Dom/ElementDump.php on line 55
[06-Apr-2021 08:46:38 UTC] PHP Notice:  Undefined property: AmpProject\Dom\Element::attributes in /app/public/content/plugins/amp/vendor/ampproject/amp-toolbox/src/Dom/Element.php on line 100
[06-Apr-2021 08:46:38 UTC] PHP Fatal error:  Method AmpProject\Dom\ElementDump::__toString() must not throw an exception, caught TypeError: Argument 1 passed to iterator_to_array() must implement interface Traversable, null given in /app/public/content/plugins/amp/vendor/ampproject/amp-toolbox/src/Optimizer/Error/CannotPreloadImage.php on line 0
[06-Apr-2021 08:46:38 UTC] PHP Fatal error:  Unknown: Cannot use output buffering in output buffering display handlers in Unknown on line 0
[06-Apr-2021 08:49:25 UTC] PHP Warning:  Couldn't fetch AmpProject\Dom\Element. Node no longer exists in /app/public/content/plugins/amp/vendor/ampproject/amp-toolbox/src/Dom/ElementDump.php on line 55
[06-Apr-2021 08:49:25 UTC] PHP Notice:  Undefined property: AmpProject\Dom\Element::attributes in /app/public/content/plugins/amp/vendor/ampproject/amp-toolbox/src/Dom/Element.php on line 100
[06-Apr-2021 08:49:25 UTC] PHP Fatal error:  Method AmpProject\Dom\ElementDump::__toString() must not throw an exception, caught TypeError: Argument 1 passed to iterator_to_array() must implement interface Traversable, null given in /app/public/content/plugins/amp/vendor/ampproject/amp-toolbox/src/Optimizer/Error/CannotPreloadImage.php on line 0
[06-Apr-2021 08:49:25 UTC] PHP Fatal error:  Unknown: Cannot use output buffering in output buffering display handlers in Unknown on line 0
```

Further investigation led to the query `AmpProject\AmpWP\Transformer\DetermineHeroImages::CUSTOM_HEADER_XPATH_QUERY` incorrectly detecting the `noscript>img` due to the subquery `not( contains( concat( ' ', normalize-space( @class ), ' ' ), ' custom-logo ' ) )` now being true as the `custom-logo` class is no longer being copied.

Reply:
Fatal error prevented with fb0928f98 but I'm still not entirely clear why the error occurred in the first place.

Reply:
In any case, the change in fb0928f is key because we never want to select a `noscript > img` for SSR.

Reply:
Actually, it doesn't even make sense to select any `img` now that I think of it, because all of the sanitizers will have already run and so there should be no `img` left in the page that we need to SSR.

Reply:
Done in c72d487a9.

Reply:
> Fatal error prevented with fb0928f but I'm still not entirely clear why the error occurred in the first place.

The error occurred because the `noscript>img` was receiving the `data-hero-candidate` attribute, and `AmpProject\Optimizer\Transformer::detectImageWithAttribute()` detects it as a valid candidate because when it [retrieves](https://github.com/ampproject/amp-toolbox-php/blob/fafabbe05eb2815dbd5fe50edaba81925e763feb/src/Optimizer/Transformer/PreloadHeroImage.php#L289) the CSS background image URL for the image (which resolves to an empty string `''`) and parses it as a URL (which becomes the valid URL path `/`), it is returned as a hero image for preloading, which should not be the case.

Reply:
I understand why the error was happening: `PreloadHeroImage::findHeroImages()` was including the `noscript>img` for an `amp-img` that was going to be prerendered. So it included both the `amp-img` and the `amp-img > noscript > img` as the hero images. When prerendering occurred, then the latter got removed and so it was no longer in the document. And then the exception ensued.

This should all be sorted now.